saplings/circuits/src/components/circuitDetails/CircuitDetails.js

```diff
@@ -0,0 +1,165 @@
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faQuestionCircle,
+  faArrowLeft
+} from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { useParams, Link } from 'react-router-dom';
+import React from 'react';
+
+import { Circuit } from '../../data/circuits';
+import { useCircuitState } from '../../state/circuits';
+import { getNodeRegistry } from '../../api/splinter';
+
+const CircuitDetails = () => {
+  const { circuitId } = useParams();
+  const [circuit] = useCircuitState(circuitId);
+
+  if (!circuit) {
+    return <div />;
+  }
+  return (
+    <div>
+      <div className="main-header">
+        <div className="back-button">
+          <Link to="/circuits">
+            <FontAwesomeIcon icon={faArrowLeft} />
+            &nbsp;Circuits
+          </Link>
+        </div>
+        <div className="circuits-title">
+          <h4>
+            Circuit
+            {circuitId}
+          </h4>
+          <div className="managementType">
+            Gameroom
+            <span>
+              <FontAwesomeIcon icon={faQuestionCircle} />
+            </span>
+          </div>
+        </div>
+      </div>
+      <div className="main-content">
+        <div className="midContent">
+          <div className="circuit-stats">
+            <div className="stat total-circuits">
+              <span className="stat-count circuits-count">
+                {circuit.members.length}
+              </span>
+              Nodes
+            </div>
+            <div className="stat action-required">
+              <span className="stat-count action-required-count">
+                {circuit.roster.length}
+              </span>
+              Services
+            </div>
+          </div>
+        </div>
+
+        <NodesTable circuit={circuit} nodes={circuit.members} />
+      </div>
+    </div>
+  );
+};
+
+const NodesTable = ({ circuit, nodes }) => {
+  const [fullNodes, setNodes] = React.useState(null);
+
+  React.useEffect(() => {
+    const fetchNodes = async () => {
+      try {
+        const apiNodes = await getNodeRegistry();
+        const filteredNodes = apiNodes.filter(
+          node => !!nodes.find(id => id === node.identity)
+        );
+        setNodes(filteredNodes);
+      } catch (e) {
+        throw Error(`Unable to fetch nodes from the node registry: ${e}`);
+      }
+    };
+
+    fetchNodes();
+  }, [nodes]);
+
+  if (fullNodes === null) {
+    return <div />;
+  }
+
+  let rows = [
+    <tr>
+      <td colSpan="5" className="no-nodes-msg">
+        No Nodes found for this circuit
+      </td>
+    </tr>
+  ];
+
+  if (fullNodes.length > 0) {
+    rows = fullNodes.map(node => {
+      return (
+        <tr className="table-row">
+          <td>{node.identity}</td>
+          <td>{node.displayName}</td>
+          <td>
+            {node.metadata.company || node.metadata.organization || 'N/A'}
+          </td>
+          <td>{node.endpoints[0] || 'N/A'}</td>
```

Comment:
Is there are reason to display only one endpoint?

Reply:
Other than there being no example with multiples, no.  I could list them out, each on a new line.